src/main/java/TaskManager.java

```diff
@@ -0,0 +1,239 @@
+            Duke.printDivider();
+        } catch (IOException e) {
+            Duke.printlnTab("☹ OOPS!!! Error writing to data file");
+
+        }
+    }
+
+    private String[] getTaskDetails(TaskEnum taskType, String userInputWithoutTaskCommand) throws IncompleteInformationException {
+        String[] taskDetails;
+        //strip userInputWithoutTaskCommand prevents empty description / dates
+        if (taskType == TaskEnum.DEADLINE) {
+            taskDetails = userInputWithoutTaskCommand.strip().split("/by");
+        } else { //if EVENT
+            taskDetails = userInputWithoutTaskCommand.strip().split("/at");
+        }
+
+        // taskDetails[] should have length of 2
+        //containing Task description (index 0) and Task date (index 1)
+        //special case of length 2 when "/by timing" which is still invalid
+        //is checked by .isBlank()
+        if (taskDetails.length != 2
+                || taskDetails[TASK_DESCRIPTION_INDEX].isBlank()
+                || taskDetails[TASK_DATE_INDEX].isBlank()) {
+            throw new IncompleteInformationException();
+        }
+
+        for (int i = 0; i < 2; i++) {
+            taskDetails[i] = taskDetails[i].strip();
+        }
+        return taskDetails;
+    }
+
+    public void addTodo(String description, boolean isDone) {
+        tasks.add(new Todo(description, isDone));
+    }
+
+    public void addDeadline(String[] taskDetails, boolean isDone) {
+        tasks.add(new Deadline(taskDetails[TASK_DESCRIPTION_INDEX], isDone, taskDetails[TASK_DATE_INDEX]));
+    }
+
+    public void addEvent(String[] taskDetails, boolean isDone) {
+        tasks.add(new Event(taskDetails[TASK_DESCRIPTION_INDEX], isDone, taskDetails[TASK_DATE_INDEX]));
+    }
+
+    private void addTaskSuccess() {
+        Duke.printlnTab("Got it. I've added this task:");
+        Duke.printlnTab(" " + tasks.get(tasks.size() - 1)); //latest item
+        printNumberOfTasksMessage();
+        Duke.printDivider();
+    }
+
+    public void deleteTask(String userInput) throws BlankDescriptionException, ExceedTotalTasksException {
+        String taskNumberStr = userInput.substring(DELETE_NUMBER_INDEX).strip();
+        if (taskNumberStr.isBlank()) {
+            throw new BlankDescriptionException();
+        }
+
+        //taskNumber displayed starting with 1
+        //but array starts with 0
+        int taskNumber = Integer.parseInt(taskNumberStr);
+        if (taskNumber > tasks.size()) {
+            throw new ExceedTotalTasksException();
+        }
+
+        int taskIndex = taskNumber - 1;
+
+        String taskToStringOutput = String.format(" %s", tasks.get(taskIndex));
+
+        tasks.remove(taskIndex);
+
+        Duke.printlnTab("Noted. I've removed this task:");
+        Duke.printlnTab(taskToStringOutput);
+
+        printNumberOfTasksMessage();
+        Duke.printDivider();
+    }
+
+    private void doneTask(String userInput) throws BlankDescriptionException, ExceedTotalTasksException {
```

Comment:
Generally you could provide method headers that describe what the method is doing to improve readability. There are piece-wise comments that do not give the reader an overall picture of what is happening.